src/Datadog.Trace/ScopeManagerBase.cs

```diff
@@ -31,23 +31,22 @@ public Scope Activate(Span span, bool finishOnClose)
         {
             var newParent = Active;
             var scope = new Scope(newParent, span, this, finishOnClose);
-            var scopeOpenedArgs = new SpanEventArgs(span);
 
-            if (newParent == null)
+            if (newParent == null && TraceStarted != null)
             {
-                TraceStarted?.Invoke(this, scopeOpenedArgs);
+                TraceStarted(this, new SpanEventArgs(span));
```

Comment:
That's a bad practice. You're introducing a subtle race condition if TraceStarted becomes null between line 35 and line 37

Reply:
Makes sense, I really don't remember why I changed, I think I was testing because the original only adds a `dup` opcode to use the same value... reverting it...

Reply:
done in b3feead